File: examples/go/blake3/Blake3Test_test.go
package contract

import (
	"testing"
	runar "github.com/icellan/runar/packages/runar-go"
)

func newBlake3Test() *Blake3Test {
	// Mock BLAKE3 functions return 32 zero bytes, so set Expected to match.
	return &Blake3Test{
		Expected: runar.ByteString(make([]byte, 32)),
	}
}

func TestBlake3Test_VerifyCompress(t *testing.T) {
	c := newBlake3Test()
	chainingValue := runar.ByteString(make([]byte, 32))
	block := runar.ByteString(make([]byte, 64))
	c.VerifyCompress(chainingValue, block)
}

func TestBlake3Test_VerifyHash(t *testing.T) {
	c := newBlake3Test()
	message := runar.ByteString(make([]byte, 32))
	c.VerifyHash(message)
}

func TestBlake3Test_Compile(t *testing.T) {
	if err := runar.CompileCheck("Blake3Test.runar.go"); err != nil {
		t.Fatalf("Rúnar compile check failed: %v", err)
	}
}

File: examples/rust/blake3/Blake3Test_test.rs
#[path = "Blake3Test.runar.rs"]
mod contract;

use contract::*;

fn new_blake3_test() -> Blake3Test {
    // Mock BLAKE3 functions return 32 zero bytes, so set expected to match.
    Blake3Test {
        expected: vec![0u8; 32],
    }
}

#[test]
fn test_verify_compress() {
    let chaining_value = vec![0u8; 32];
    let block = vec![0u8; 64];
    new_blake3_test().verify_compress(&chaining_value, &block);
}

#[test]
fn test_verify_hash() {
    let message = vec![0u8; 32];
    new_blake3_test().verify_hash(&message);
}

#[test]
fn test_compile() {
    runar::compile_check(include_str!("Blake3Test.runar.rs"), "Blake3Test.runar.rs").unwrap();
}
